index/matchtree.go

```diff
@@ -1435,3 +1468,20 @@ func isRegexpAll(r *syntax.Regexp) bool {
 		return false
 	}
 }
+
+func queryMetaChecksum(field string, value *regexp.Regexp) string {
+	h := xxhash.New()
+	h.Write([]byte(field))
+	h.Write([]byte{':'})
+	h.Write([]byte(value.String()))
+	return fmt.Sprintf("%x", h.Sum64())
+}
+
+func queryRepoIdsChecksum(repos []zoekt.Repository) string {
+	h := xxhash.New()
+	for _, r := range repos {
+		h.Write([]byte(fmt.Sprint(r.ID)))
+		h.Write([]byte{','})
+	}
+	return fmt.Sprintf("%x", h.Sum64())
```

Comment:
can't help myself to write an implementation that avoids intermediate allocations :)

```suggestion
	h := xxhash.New()
	var idBuf [10]byte // 10 max len of uint32 string
	for _, r := range repos {
		h.Write(strconv.AppendUint(idBuf[:0], uint64(r.ID), 10))
		h.Write([]byte{','})
	}
	return fmt.Sprintf("%x", h.Sum64())
```